### PR TITLE
feat: add uncommitted option

### LINKED ISSUE #217: Support uncommitted option for style linting
I wold love to use this in my pre-commit hook, currently it supports affected which is good for CI. 

Can we add support for uncommitted option 
https://github.com/nrwl/nx/blob/b9ef28712d0d23b0e575a4c354d1bf438e3e6f34/packages/workspace/src/command-line/shared.ts#L12

### PR DESCRIPTION
**Pull request by santoshyadavdev**
Closes #217

### CODE REVIEW

**Comment by Phillip9587** [comment]
@santoshyadavdev You have to filter the uncomitted files with the `lintFilePatterns` to ensure that only files that are listed in the `lintFilePatterns` and uncommitted are passed to stylelint.  In your current implementation all uncomitted files are passed to stylelint which leads to unexpected behaviour.

**Comment by santoshyadavdev** [comment]
Wait don't review, this is just draft, I pushed code which I could work last weekend. Will continue this weekend.

File: packages/nx-stylelint/src/executors/lint/executor.ts [hunk 1]
```diff
@@ -70,4 +75,27 @@ export async function lintExecutor(
...
+
+function getUncommittedFiles(format: string): string[] {
+  return parseGitOutput(`git diff --name-only --relative HEAD .`, format);
+}
+
+function getUntrackedFiles(format: string): string[] {
+  return parseGitOutput(`git ls-files --others --exclude-standard`, format);
+}
+
+function parseGitOutput(command: string, format: string): string[] {
```
Comment (santoshyadavdev) [inline packages/nx-stylelint/src/executors/lint/executor.ts:91]: Hey @Phillip9587 ,

one question just realized lintFilePatterns is an array, to filter files I was thinking about using minimatch like below to get the files, is it ok to introduce minimatch, or should I look for other options.

`return format.map((pattern) => minimatch.match(files, pattern, { matchBase: true })).flat(); `

Comment (Phillip9587) [reply packages/nx-stylelint/src/executors/lint/executor.ts:91]: Just use minimatch

**Comment by santoshyadavdev** [comment]
Update on this PR , e2e test is pending looking to wrap this weekend.

**Comment by trystan2k** [comment]
Hi. 

I just found this amazing tool for use stylelint with NX workspaces and I was just looking if this feature (lint only staged files) was available and found this PR. 

Is this already finished ? Can it be merged ?

Thanks!

**Comment by santoshyadavdev** [comment]
Unfortunately I migrated to a new country and am having a hard time completing the test cases and some edge cases. But I plan to complete it next month once I settle down.

**Comment by Phillip9587** [comment]
@santoshyadavdev Sorry I'm gonna close this for now. Feel free to open a new PR when it's ready for review.

**Comment by MAN-Sendance** [comment]
It would be really nice to get this
